src/widgets/conditions/condition-record.component.tsx

```diff
@@ -1,39 +1,77 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { useRouteMatch } from "react-router-dom";
 import { getConditionByUuid } from "./conditions.resource";
 import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { useCurrentPatient, newWorkspaceItem } from "@openmrs/esm-api";
 import dayjs from "dayjs";
 import styles from "./condition-record.css";
 import SummaryCard from "../../ui-components/cards/summary-card.component";
-import { useCurrentPatient } from "@openmrs/esm-api";
+import { ConditionsForm } from "./conditions-form.component";
 
 export default function ConditionRecord(props: ConditionRecordProps) {
-  const [patientCondition, setPatientCondition] = React.useState(null);
+  const [patientCondition, setPatientCondition] = useState(null);
   const [isLoadingPatient, patient, patientUuid] = useCurrentPatient();
   const match = useRouteMatch();
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (!isLoadingPatient && patient) {
       const sub = getConditionByUuid(match.params["conditionUuid"]).subscribe(
-        condition => setPatientCondition(condition),
+        ({ resource }) => setPatientCondition(resource),
         createErrorHandler()
       );
       return () => sub.unsubscribe();
     }
   }, [isLoadingPatient, patient, match.params]);
 
+  const openEditConditionsWorkspaceTab = (
+    componentName,
+    title,
+    conditionUuid,
+    name,
+    clinicalStatus,
+    onsetDateTime
+  ) => {
+    newWorkspaceItem({
+      component: componentName,
+      name: title,
+      props: {
+        match: {
+          params: {
+            conditionUuid,
+            name,
+            clinicalStatus,
+            onsetDateTime
+          }
+        }
+      },
+      inProgress: false,
+      validations: (workspaceTabs: any[]) =>
```

Comment:
Why `any`? Do we have a type here? If you expect the `component` prop to be there then at least type it as `{ component: string }` to communicate that requirement.

Reply:
Given that it workspace tabs seems to expect an array of function components, do you think the appropriate type would be `Array<{ component: React.FunctionComponent }>`?

This pattern where widgets have their own specific tab launcher functions has been repeated in many places across the app - here Conditions has its own variant - `openConditionsWorkspaceTab`, Allergies has `openAllergiesWorkspaceTab` and so on. It's probably a good candidate for DRYing up.

Reply:
So `componentName` refers to a  `React.FC`? Sure - I would also recommend renaming `componentName` to `component` to reflect this. Usually, a name is a string (at least to me).

Rest I fully agree! :+1:

Reply:
No, `component` refers to a `React.FC`. `componentName` remains a string - the name of the component.

```
// workspace.resource.tsx (from openmrs-esm-api)
export function newWorkspaceItem(item: WorkspaceItem) {
  workspaceItem.next(item);
}

export type WorkspaceItem = {
  component: any;
  name: string;
  props: any;
  validations?: Function;
  inProgress: boolean;
  componentClosed?: Function;
};
```

Reply:
> Given that it workspace tabs seems to expect an array of function components, do you think the appropriate type would be `Array<{ component: React.FunctionComponent }>`?
> 
> This pattern where widgets have their own specific tab launcher functions has been repeated in many places across the app - here Conditions has its own variant - `openConditionsWorkspaceTab`, Allergies has `openAllergiesWorkspaceTab` and so on. It's probably a good candidate for DRYing up.

#47 addresses the issue of making the newWorkspaceItem logic reusable.